Dockerfile

```diff
@@ -0,0 +1,51 @@
+# Use the jupyter/minimal-notebook as the base image
+FROM quay.io/jupyter/minimal-notebook:latest
+
+# Metadata labels
+LABEL org.opencontainers.image.title="Python Tutorial"
+LABEL org.opencontainers.image.description="A containerized Python tutorial environment with Jupyter Lab."
+LABEL org.opencontainers.image.authors="Empa Scientific IT <scientificit@empa.ch>"
+LABEL org.opencontainers.image.url="https://github.com/empa-scientific-it/python-tutorial"
+LABEL org.opencontainers.image.source="https://github.com/empa-scientific-it/python-tutorial"
+LABEL org.opencontainers.image.version="1.0.0"
+LABEL org.opencontainers.image.licenses="MIT"
+
+# Set environment variables for the tutorial and repository
+ENV BASENAME="python-tutorial"
+ENV REPO=${HOME}/${BASENAME}
+ENV IPYTHONDIR="${HOME}/.ipython"
+
+# Switch to root user to install additional dependencies
+USER root
+RUN apt-get update && \
+    apt-get install -y --no-install-recommends \
+    build-essential \
+    gcc \
+    g++ \
+    libffi-dev && \
+    apt-get clean && \
+    rm -rf /var/lib/apt/lists/*
+
+# Switch back to the default notebook user
+USER ${NB_UID}
+
+# Set up the Conda environment
+COPY docker/environment.yml /tmp/environment.yml
+RUN mamba env update -n base -f /tmp/environment.yml && \
+    mamba clean --all -f -y && \
+    fix-permissions "${CONDA_DIR}" && \
+    fix-permissions "/home/${NB_USER}"
+
+# Prepare IPython configuration (move earlier in the build)
+RUN mkdir -p ${HOME}/.ipython/profile_default
+COPY --chown=${NB_UID}:${NB_GID} binder/ipython_config.py ${HOME}/.ipython/profile_default/
+
+# Copy the repository late in the build process
+RUN mkdir -p ${REPO}
+COPY --chown=${NB_UID}:${NB_GID} . ${REPO}/
```

Comment:
This part is a bit tricky. The user should either start the **same** container (so the readme needs to be updated), or we should mount the folder to the container.

In any case, the readme should be synchronized with the new approach.

Reply:
I think the step is correct and makes sense only when we're building the image. The user can still mount the local folder with a local copy of the repository, and nothing changes if the are no mismatches. Otherwise they will have their local repository inside the container.

Reply:
It might be good to create a docker volume, but I have to think about it carefully.

Reply:
I did a bit of research. It seems that those lines **are not needed**: I added them to make sure to have the repository material inside the container, but:

1. If the user is mounting a local folder with the repository, that's redundant
2. GitHub Codespaces, which can use a prebuilt Docker image to speed up the startup, also clones the repository by default, so the lines are again redundant

If you agree that this is the case, we can remove them and speed up building the image.

Reply:
I suggest we have a quick chat about this today during the meeting, and then we can decide.